### PR TITLE
[SPARK-7384][Core][Tests] Fix flaky tests for distributed mode in BroadcastSuite

### PR DESCRIPTION
**Pull request by zsxwing**
Fixed the following failure: https://amplab.cs.berkeley.edu/jenkins/job/Spark-1.3-Maven-pre-YARN/hadoop.version=1.0.4,label=centos/452/testReport/junit/org.apache.spark.broadcast/BroadcastSuite/Unpersisting_HttpBroadcast_on_executors_and_driver_in_distributed_mode/

The tests should wait until all slaves are up. Otherwise, there may be only a part of `BlockManager`s registered, and fail the tests.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/broadcast/BroadcastSuite.scala [hunk 1]
```diff
@@ -17,9 +17,12 @@
 
 package org.apache.spark.broadcast
 
+import scala.concurrent.duration._
+import scala.language.postfixOps
```
Comment (rxin) [inline core/src/test/scala/org/apache/spark/broadcast/BroadcastSuite.scala]: maybe get rid of this and just write 10.seconds

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 2]
```diff
@@ -407,15 +407,15 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
     if (master == "yarn-client") System.setProperty("SPARK_YARN_MODE", "true")
 
+    _jobProgressListener = new JobProgressListener(_conf)
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: document inline why we moved this

**Comment by rxin** [comment]
LGTM.

**Comment by rxin** [comment]
Merging in master. Thanks.

**Comment by zsxwing** [comment]
I think it's better to merge this one to 1.4, too. Right?

**Comment by rxin** [comment]
I actually did.

**Comment by zsxwing** [comment]
Sorry. I thought it wasn't in 1.4 because JIRA wasn't closed. I just closed the JIRA.

**Pull request closed without merging**